### PR TITLE
High low scores

### LINKED ISSUE #11: lowest_scoring_home_team
Name of the team with the lowest average score per game across all seasons when they are at home.

String

### LINKED ISSUE #24: highest_scoring_visitor
Name of the team with the highest average score per game across all seasons when they are away.

String

### LINKED ISSUE #27: highest_scoring_home_team
Name of the team with the highest average score per game across all seasons when they are home.

String

### LINKED ISSUE #31: lowest_scoring_visitor
Name of the team with the lowest average score per game across all seasons when they are a visitor.

String

### PR DESCRIPTION
**Pull request by kcockett**
Created high/low score comparisons.   Included a big helper method.  All tests passed with 100% coverage.

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -93,21 +97,61 @@ def average_goals_per_game
...
+  
+  def lowest_scoring_home_team 
+    tally_scores = Hash.new {|h, k| h[k] = [] }
+    games.each do |game|
+      tally_scores[game.home_team_id] <<  game.home_goals.to_i
+    end
+    average_and_lookup(tally_scores, :min)
+  end
+
+  ### HELPER FUNCTION ###
```
Comment (WZale) [inline lib/stat_tracker.rb:145]: I like the helper method! It made me realize how we could refactor our best and worst offense methods, which is good to see.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -93,21 +97,61 @@ def average_goals_per_game
   end
   
   def average_goals_by_season
-    hash = {}
+    all_goals = Hash.new {|h, k| h[k] = [] }
```
Comment (WZale) [inline lib/stat_tracker.rb:101]: Look at that hash syntax! That's awesome. Super clean.

**Review by Gabe-Torres — APPROVED** [review]

**Review by WZale — APPROVED** [review]
You two are really making things happen over there. Well done!

**Review by Gabe-Torres — APPROVED** [review]
Looks good y'all! I like the refactors

**Pull request merged**